### PR TITLE
Don't add csv or excel in dropdown if portphp/csv or excel lib is not…

### LINKED ISSUE #155: Dropdowns for format of export/import when portphp/csv portphp/excel have not been required
When portphp/csv portphp/excel have not been required, the CSV and Excel a steal proposed in the dropdown, it lead to an error 500 , because the services for theses format are not registred.

### PR DESCRIPTION
**Pull request by spras**
fix #155 

Don't know if this is the right way  to fix it ?

### CODE REVIEW

**Comment by lsmith77** [comment]
might make more sense to drive this from the DI layer so that we can run this logic once in the extension while building the DI container.

**Comment by stefandoorn** [comment]
I think the respective services simply shouldn't be loaded, therefore the registering of types will not find these tagged services. Didn't we have this in place yet, e.g. here https://github.com/FriendsOfSylius/SyliusImportExportPlugin/blob/master/src/DependencyInjection/FOSSyliusImportExportExtension.php#L53.

**Comment by spras** [comment]
The csv/excel service aren't loaded, and this cause an error when you try to export as csv format => only json exporter are registred. This is possible to try to export to csv format because the dropdown propose csv, that's what i try to avoid.    

I agree that this logic should be run once. Perhaps in the formType for export we should rely on registred services ?

**Comment by lsmith77** [comment]
looks good to me .. @solutionDrive-Alt wdyt?

File: src/Form/ImportType.php [hunk 1]
```diff
@@ -4,22 +4,46 @@
...
+    }
+
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        /** @var string $importerType */
+        $importerType = $options['importer_type'];
+        /** @var \FriendsOfSylius\SyliusImportExportPlugin\Exporter\ExporterRegistry $importerRegistry */
+        $importerRegistry = $options['importer_registry'];
+
+        $choices = [];
```
Comment (stefandoorn) [inline src/Form/ImportType.php]: My preference would be to offload this to a private function `getChoices`, to keep the `buildForm` method itself small.

File: src/Form/ImportType.php [hunk 2]
```diff
@@ -4,22 +4,46 @@
...
+        $csvImporter = ExporterRegistry::buildServiceName($importerType, 'csv');
```
Comment (stefandoorn) [inline src/Form/ImportType.php]: Saving it in a var is not really needed here I think, it's only used a single time afterwards. Same goes for the `$xlsxImporter` var.

Comment (spras) [reply src/Form/ImportType.php]: of course, nicer

File: src/Form/ImportType.php (hunk 1 above)
Comment (spras) [reply src/Form/ImportType.php]: yes, much better

File: src/Form/ImportType.php [hunk 3]
```diff
@@ -4,22 +4,46 @@
...
 
 class ImportType extends AbstractType
 {
+    private const CLASS_CSV_READER = CsvReaderFactory::class;
+    private const CLASS_EXCEL_READER = ExcelReaderFactory::class;
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setRequired('importer_type');
+        $resolver->setRequired('importer_registry');
```
[the remark this answers is not in the document]
Comment (spras) [reply src/Form/ImportType.php]: indeed, we can inject the importerRegistry

File: src/Form/ImportType.php [hunk 4]
```diff
@@ -4,22 +4,46 @@
...
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 class ImportType extends AbstractType
 {
+    private const CLASS_CSV_READER = CsvReaderFactory::class;
+    private const CLASS_EXCEL_READER = ExcelReaderFactory::class;
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setRequired('importer_type');
```
[the remark this answers is not in the document]
Comment (spras) [reply src/Form/ImportType.php]: Sorry i don't understand what you meant ?

[the remark this answers is not in the document]
Comment (stefandoorn) [reply src/Form/ImportType.php]: Besides saying that `importer_type` is required, can't we also define that it should be a `string`? I thought that's possible using `OptionsResolver`.

[the remark this answers is not in the document]
Comment (spras) [reply src/Form/ImportType.php]: ok i see !

**Comment by stefandoorn** [comment]
Only some tests are failing now :-(

**Comment by lsmith77** [comment]
I guess the service definition needs to be updated to also remove the `@router`
https://github.com/FriendsOfSylius/SyliusImportExportPlugin/blob/master/src/Resources/config/services.yml#L8

**Comment by lsmith77** [comment]
this PR needs to be updated due to https://github.com/FriendsOfSylius/SyliusImportExportPlugin/pull/169

should hopefully just be search and replace

**Comment by spras** [comment]
updated due to #169, remove the `@router` in the service definition, and now test are OK

**Comment by lsmith77** [comment]
thank you so much! great work.

**Pull request merged**